src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/alerts/AlertsSynchronisationIntTest.kt

```diff
@@ -934,86 +934,188 @@ class AlertsSynchronisationIntTest : SqsIntegrationTestBase() {
+      }
+
+      @Test
+      fun `will send missing alerts to DPS`() {
+        dpsAlertsServer.verify(
+          postRequestedFor(urlPathEqualTo("/merge/A1234KT/alerts"))
+            .withRequestBodyJsonPath("$[0].offenderBookId", "$bookingId")
+            .withRequestBodyJsonPath("$[0].alertSeq", "3")
+            .withRequestBodyJsonPath("$[1].offenderBookId", "$bookingId")
+            .withRequestBodyJsonPath("$[1].alertSeq", "4"),
+        )
+      }
 
-    @Test
-    fun `will create a mapping between the DPS and NOMIS alerts`() {
-      alertsMappingApiMockServer.verify(
-        postRequestedFor(urlPathEqualTo("/mapping/alerts/batch"))
-          .withRequestBodyJsonPath("$[0].nomisBookingId", "$bookingId")
-          .withRequestBodyJsonPath("$[0].nomisAlertSequence", "3")
-          .withRequestBodyJsonPath("$[0].dpsAlertId", "$dpsAlertId1")
-          .withRequestBodyJsonPath("$[1].nomisBookingId", "$bookingId")
-          .withRequestBodyJsonPath("$[1].nomisAlertSequence", "4")
-          .withRequestBodyJsonPath("$[1].dpsAlertId", "$dpsAlertId2"),
-      )
+      @Test
+      fun `will create a mapping between the DPS and NOMIS alerts`() {
+        alertsMappingApiMockServer.verify(
+          postRequestedFor(urlPathEqualTo("/mapping/alerts/batch"))
+            .withRequestBodyJsonPath("$[0].nomisBookingId", "$bookingId")
+            .withRequestBodyJsonPath("$[0].nomisAlertSequence", "3")
+            .withRequestBodyJsonPath("$[0].dpsAlertId", "$dpsAlertId1")
+            .withRequestBodyJsonPath("$[1].nomisBookingId", "$bookingId")
+            .withRequestBodyJsonPath("$[1].nomisAlertSequence", "4")
+            .withRequestBodyJsonPath("$[1].dpsAlertId", "$dpsAlertId2"),
+        )
+      }
+
+      @Test
+      fun `will track telemetry for the merge`() {
+        verify(telemetryClient).trackEvent(
+          eq("from-nomis-synch-alerts-merge"),
+          check {
+            assertThat(it["bookingId"]).isEqualTo(bookingId.toString())
+            assertThat(it["offenderNo"]).isEqualTo("A1234KT")
+            assertThat(it["removedOffenderNo"]).isEqualTo("A1000KT")
+            assertThat(it["newAlertsCount"]).isEqualTo("2")
+            assertThat(it["newAlerts"]).isEqualTo("3, 4")
+          },
+          isNull(),
+        )
+      }
     }
 
-    @Test
-    fun `will track telemetry for the merge`() {
-      verify(telemetryClient).trackEvent(
-        eq("from-nomis-synch-alerts-merge"),
-        check {
-          assertThat(it["bookingId"]).isEqualTo(bookingId.toString())
-          assertThat(it["offenderNo"]).isEqualTo("A1234KT")
-          assertThat(it["removedOffenderNo"]).isEqualTo("A1000KT")
-          assertThat(it["newAlertsCount"]).isEqualTo("2")
-          assertThat(it["newAlerts"]).isEqualTo("3, 4")
-        },
-        isNull(),
-      )
+    @Nested
+    inner class HappyPathWithFailure {
```

Comment:
Doesn't sound very happy?

Reply:
We eventually it works so it happy in the end.
About to add a new called UnHappy